### PR TITLE
update k8s deps to v1.11

### PR DESCRIPTION
**Pull request by ayj**
This PR updates our k8s package dependencies to v1.11 in preparation for the 1.1 release. Rationale for selecting v1.11 is below:

* Istio components depends on k8s.io repos/packages for k8s integration, e.g. API types, clients, controllers. These dependencies are versioned against upstream k8s releases.

* k8s' own [supported releases and component skew](https://github.com/kubernetes/client-go#compatibility-client-go---kubernetes-clusters) says "_k8s clients should be skewed no more than one minor version from the master, but may lead the master by up to one minor version_".

* While [k8s/client-go's compatibility matrix](https://github.com/kubernetes/client-go#compatibility-client-go---kubernetes-clusters) describes a stronger backwards compatibility against older k8s cluster versions, other k8s.io repos (e.g. k8s.io/api, k8s.io/api, k8s.io/apimachinery) do not.

* Istio supports three most recent k8s minor releases. When Istio 1.1.x is released, k8s v1.12 will be most latest minor release and v1.12 will be the minimum supported version.

* v1.11 plus one minor version is v1.12 (max supported). v1.11 minus one minor version is v1.10 (min supported).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,6 +37,7 @@ initWorkingDir: &initWorkingDir
     mkdir -p /go/out/tests
     mkdir -p /go/out/logs
     mkdir -p /home/circleci/logs
+    curl https://dl.google.com/go/go1.10.4.linux-amd64.tar.gz | sudo tar xz -C $(go env GOROOT) --strip-components=1
```
Comment (costinm) [inline .circleci/config.yml]:
```suggestion
    # Temporary: until "machine" image is updated or customized.
    curl https://dl.google.com/go/go1.10.4.linux-amd64.tar.gz | sudo tar xz -C $(go env GOROOT) --strip-components=1
```

File: mixer/pkg/server/monitoring.go [hunk 2]
```diff
@@ -61,7 +59,7 @@ func startMonitor(port uint16, enableProfiling bool, lf listenFunc) (*monitor, e
 	mux := http.NewServeMux()
 
 	registry := prometheus.NewRegistry()
-	registry.MustRegister(prometheus.NewProcessCollector(os.Getpid(), ""))
+	registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
```
Comment (costinm) [inline mixer/pkg/server/monitoring.go]: Why ? @douglas-reid

Comment (douglas-reid) [reply mixer/pkg/server/monitoring.go]: does the client_golang update here match https://github.com/istio/istio/pull/9342 ?

Comment (ayj) [reply mixer/pkg/server/monitoring.go]: Yes. For whatever reason, Gopkg.lock was at version 0.9.0 while the Gopkg.toml constraint is 0.9.0-pre1. I updated Gopkg.toml to match #9342 and make the dep files consistent.

Comment (ayj) [reply mixer/pkg/server/monitoring.go]: `prometheus.NewProcessCollector` was [added](https://github.com/istio/istio/pull/7989/files#diff-804f2a95859b6e40df217d24109e985bR64) in the master branch when mixer self-monotiring was switched over to OpenCensus.

**Pull request merged**